packages/plugin/src/plugin.ts

```diff
@@ -0,0 +1,198 @@
+import type * as nexus from 'nexus-rpc';
+import type { DataConverter } from '@temporalio/common';
+import {
+  ClientInterceptors,
+  ClientOptions,
+  ClientPlugin,
+  ConnectionPlugin,
+  ConnectionOptions,
+  WorkflowClientInterceptors,
+  WorkflowClientInterceptor,
+} from '@temporalio/client';
+import type {
+  BundlerPlugin,
+  NativeConnectionPlugin,
+  WorkerInterceptors,
+  WorkerPlugin,
+  WorkflowBundleOption,
+  WorkerOptions,
+  ReplayWorkerOptions,
+  Worker,
+  BundleOptions,
+  NativeConnectionOptions,
+  TLSConfig,
+} from '@temporalio/worker';
+
+type PluginParameter<T> = T | ((p: T | undefined) => T);
+
+export interface SimplePluginOptions {
+  readonly name: string;
+  readonly tls?: PluginParameter<TLSConfig | boolean | null>;
+  readonly apiKey?: PluginParameter<string>;
+  readonly dataConverter?: PluginParameter<DataConverter>;
```

Comment:
We should have a custom resolve handler for `DataConverters` that merge each of the three subproperties independently. Under its present form, it would be essentially to set this property as an immediate value.